OWML.ModHelper.Interaction/ModInteraction.cs

```diff
@@ -64,10 +72,26 @@ public IModBehaviour GetMod(string uniqueName)
             return _modList.First(m => m.ModHelper.Manifest.UniqueName == uniqueName);
         }
 
-        public T GetMod<T>(string uniqueName) where T : IModBehaviour
+        private object GetApi(string uniqueName)
         {
             var mod = GetMod(uniqueName);
-            return (T)mod;
+            return mod.Api;
+        }
+
+        public TInterface GetModApi<TInterface>(string uniqueName) where TInterface : class
+        {
+            var inter = GetApi(uniqueName);
+            if (inter == null)
+            {
+                return null;
+            }
+
+            if (inter is TInterface castInter)
+            {
+                return castInter;
+            }
+
+            return _proxyFactory.CreateProxy<TInterface>(inter, _modManifest.UniqueName, uniqueName);
```

Comment:
What's the difference between _modManifest.UniqueName and uniqueName? If they're the same, the manifest could be removed?

Reply:
ones for the mod using the api, ones for the mod providing the api (i think)

Reply:
probably should rename uniqueName

Reply:
Hm, why would both be needed? 🤔 for logging/errors?